### PR TITLE
[TextFields] Corrections to height and frame and better manual layout support

### LINKED ISSUE #1517: [TextFields] Underline constraint does not take underline label content into account
### Overview
The input controller's underlineY constraint is too high when TAMIC is on. It needs to check for underline leading and trailing content to calculate. 

### Reproduction steps
 Add a TextField with floating controller and TAMIC on.

### Related issues
#1487

### PR DESCRIPTION
**Pull request by willlarche**
Fixes #1517 and #1488

### CODE REVIEW

File: catalog/MDCCatalog.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -593,7 +601,6 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				36683DCA1EF3262F002EF2E6 /* MDCEarlGreyTests.swift in Sources */,
```
Comment (randallli) [inline catalog/MDCCatalog.xcodeproj/project.pbxproj]: does the earl gray stuff belong in a different pr?

File: components/TextFields/examples/TextFieldManualLayoutExample.m [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  [self.view addSubview:textFieldName];
+
+  textFieldName.placeholder = @"Full Name";
+  textFieldName.delegate = self;
+  textFieldName.clearButtonMode = UITextFieldViewModeUnlessEditing;
+
+  self.nameController = [[MDCTextInputController alloc] initWithTextInput:textFieldName];
+
+  self.nameController.presentationStyle = MDCTextInputPresentationStyleFloatingPlaceholder;
+  self.nameController.textInput.frame = CGRectMake(10, 40, CGRectGetWidth(self.view.bounds)-20, 0);
```
Comment (randallli) [inline components/TextFields/examples/TextFieldManualLayoutExample.m]: it may read easier to set the frame of the textFieldName.frame = ...

File: components/TextFields/examples/TextFieldManualLayoutExample.m [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+  self.phoneController.helperText = @"XXX-XXX-XXXX";
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+
+  [self.nameController.textInput sizeToFit];
+  [self.phoneController.textInput sizeToFit];
+  self.phoneController.textInput.frame = CGRectMake(10, CGRectGetMaxY(self.nameController.textInput.frame) + 20, CGRectGetWidth(self.view.bounds)-20, CGRectGetHeight(self.phoneController.textInput.frame));
```
Comment (randallli) [inline components/TextFields/examples/TextFieldManualLayoutExample.m:64]: setting the center would be cleaner.

**Review by randallli — APPROVED** [review]
minor nits on examples.

File: catalog/MDCCatalog.xcodeproj/project.pbxproj (hunk 1 above)
Comment (willlarche) [reply catalog/MDCCatalog.xcodeproj/project.pbxproj]: Looks like this is actually a merge mistake.

File: components/TextFields/examples/TextFieldManualLayoutExample.m (hunk 2 above)
Comment (willlarche) [inline components/TextFields/examples/TextFieldManualLayoutExample.m]: Which is our client demographic more likely to use?

File: components/TextFields/examples/TextFieldManualLayoutExample.m (hunk 3 above)
Comment (willlarche) [reply components/TextFields/examples/TextFieldManualLayoutExample.m:64]: Which is our client demographic more likely to use?

File: components/TextFields/examples/TextFieldManualLayoutExample.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/examples/TextFieldManualLayoutExample.m]: Done.

File: catalog/MDCCatalog.xcodeproj/project.pbxproj (hunk 1 above)
Comment (willlarche) [reply catalog/MDCCatalog.xcodeproj/project.pbxproj]: Done.
